### PR TITLE
Allow to access `wrapped_module` of exported models

### PR DESCRIPTION
**Pull request by frostedoyster**
This feature would be very useful to access information about the underlying model

# Contributor (creator of pull-request) checklist

 - [ ] Tests updated (for new features and bugfixes)?
 - [ ] Documentation updated (for new features)?
 - [ ] Issue referenced (for PRs that solve an issue)?

# Reviewer checklist

 - [ ] CHANGELOG updated with public API or any other important changes?

----
 📚 [Download documentation preview for this pull-request](https://nightly.link/lab-cosmo/metatensor/actions/artifacts/1760815821.zip)

### CODE REVIEW

**Comment by Luthaf** [comment]
This does not work unfortunately, since `torch.nn.Module` is not a valid TorchScript type. Can you access `exported_model._module` after loading an exported model? If yes, we could rename it to `self.wrapped_module` or something similar, and document that reading this field is fine, but writing to it is not.

**Comment by frostedoyster** [comment]
Yes, I was able to access `._module`. Let's do that then. Perhaps we should also remove `wrapped_module()` from the documentation though, since it's not accessible anymore once the model is exported

**Comment by Luthaf** [comment]
Yes, that makes sense

**Comment by Luthaf** [comment]
@frostedoyster what's the status on this? Do you still want to finish it or should we close?

File: docs/src/atomistic/reference/models/export.rst [hunk 1]
```diff
@@ -16,3 +16,4 @@ define some information about the model, using :py:class:`ModelMetadata` and
 
 .. autoclass:: metatensor.torch.atomistic.MetatensorAtomisticModel
     :members:
+    :exclude-members: wrapped_module
```
Comment (Luthaf) [inline docs/src/atomistic/reference/models/export.rst:19]:
```suggestion
```

And did you check that `module` appears in the docs?

Comment (frostedoyster) [reply docs/src/atomistic/reference/models/export.rst:19]: No, and my way of including it also seems to break torchscript

Comment (Luthaf) [reply docs/src/atomistic/reference/models/export.rst:19]: instead of `ModelInterface`, you can try to annotate with `torch.nn.Module`

**Comment by frostedoyster** [comment]
It also doesn't work. We could remove the annotation or try something more sophisticated (what though?)

**Comment by Luthaf** [comment]
Does it work without any annotation?

**Comment by Luthaf** [comment]
alternatively, you can document a member without going through a docstring, but instead adding something like

```rst
  <...> class docstring <...>
.. py:attribute:: module
    :type: ModelInterface
    
    <...> attribute docstring <...>
```

I'm not 100% sure on the syntax, I'm trying to understand https://www.sphinx-doc.org/en/master/usage/domains/python.html#directive-py-class

**Comment by frostedoyster** [comment]
No, it doesn't work without the annotation (`module=None`)

File: docs/src/atomistic/reference/models/export.rst (hunk 1 above)
Comment (Luthaf) [reply docs/src/atomistic/reference/models/export.rst:19]: This seems to work, could you add a test that we can access the module after a full export/save/load cycle? And remove the line above in the docs as well as the `wrappedmodule` function?

**Comment by frostedoyster** [comment]
It passes the tests now but when I download the docs I see wrapped_module() and I don't see the new attribute

**Pull request merged**